src/plugins/legacy/polygonRoi/readers/medContoursReader.cpp

```diff
@@ -0,0 +1,374 @@
+#include "medContoursReader.h"
+
+#include <medAbstractDataFactory.h>
+#include <medContoursWriters.h>
+#include <medTagContours.h>
+#include <medMessageController.h>
+#include <medMetaDataKeys.h>
+
+#include <vtkDataSetReader.h>
+#include <vtkDoubleArray.h>
+#include <vtkErrorCode.h>
+#include <vtkFieldData.h>
+#include <vtkIdTypeArray.h>
+#include <vtkMath.h>
+#include <vtkMetaSurfaceMesh.h>
+#include <vtkPolyData.h>
+#include <vtkPolyDataReader.h>
+#include <vtkStringArray.h>
+
+// ------------------------------------------------------------------------- //
+// ------------------------ Contour Reader Private -------------------------- //
+// ------------------------------------------------------------------------- //
+class ContoursReaderPrivate
+{
+public:
+    qint64 fileVersion;
+    bool isBinary;
+};
+
+
+// ------------------------------------------------------------------------- //
+// ---------------------------- Contour Reader ------------------------------ //
+// ------------------------------------------------------------------------- //
+medContoursReader::medContoursReader()
+    : dtkAbstractDataReader(),
+      d(new ContoursReaderPrivate())
+{
+    d->isBinary = true;
+    d->fileVersion = medContoursWriter::NUMBER_OF_EXPORT_VERSIONS;
+}
+
+medContoursReader::~medContoursReader()
+{
+    delete d;
+}
+
+bool medContoursReader::registered()
+{
+    return medAbstractDataFactory::instance()->registerDataReaderType("ContoursReader",
+                                                                      medContoursReader::s_handled(),
+                                                                      createMedContoursReader);
+}
+
+QString medContoursReader::identifier() const
+{
+    return QString("ContoursReader");
+}
+
+QString medContoursReader::description() const
+{
+    return QString("Reader for Contours (Polygon ROI)");
+}
+
+QStringList medContoursReader::handled() const
+{
+    return s_handled();
+}
+
+QStringList medContoursReader::s_handled()
+{
+    return QStringList() << "medContours";
+}
+
+QStringList medContoursReader::supportedFileExtensions() const
+{
+    return QStringList() << ".ctrb";
+}
+
+bool medContoursReader::canRead(const QString& path)
+{
+    return path.endsWith(".ctrb");
+}
+
+bool medContoursReader::canRead(const QStringList& paths)
+{
+    if (paths.empty())
+    {
+        return false;
+    }
+    return canRead(qPrintable(paths.first()));
+}
+
+bool medContoursReader::readInformation(const QString& path)
```

Comment:
Il serait préférable d'avoir un seul return

Reply:
{
    bool bRes = true;

    QFile file(path);
    if (path.endsWith(".ctrb"))
    {
        d->isBinary = true;
        // binary format, read only the header
        if (file.open(QFile::ReadOnly))
        {
            QDataStream in(&file);
            qint64 dummy = 0;
            in >> d->fileVersion >> dummy >> dummy >> dummy;
            file.close();

            if ((d->fileVersion < medContoursWriter::EXPORT_VERSION_1) ||
                (d->fileVersion >= medContoursWriter::NUMBER_OF_EXPORT_VERSIONS))
            {
                qDebug() << metaObject()->className() << ":: readInformation - unknown contours binary export version.";
                bRes = false;
            }
			else
			{
                this->setData(medAbstractDataFactory::instance()->create(QString("medContours")));
			}
        }
    }
    else
    {
        d->isBinary = false;
        // Debug ASCII format
        vtkSmartPointer<vtkDataSetReader> reader = vtkSmartPointer<vtkDataSetReader>::New();
        reader->SetFileName(qPrintable(path));
        if (reader->OpenVTKFile() == 0)
        {
            qDebug()<< metaObject()->className()
                    << "::readInformation - failed to open underlying vtk file.";
            bRes = false;
        }
		else
		{
            reader->ReadHeader();
            QString header(reader->GetHeader());
            if (!header.contains(medContoursWriter::header))
            {
                qDebug()<< metaObject()->className()
                        << "::readInformation - file does not have the correct header.";
                bRes = false;
            }
			else
			{
                this->setData(medAbstractDataFactory::instance()->create(QString("medContours")));
			}
		}
    }

    return bRes;
}

Reply:
Thx